### ISSUE TITLE
Public-facing typesafe API

### ISSUE THREAD

**Issue by vlsi** [marcphilipp/nexus-publish-plugin#25]
The plugin exposes tasks, however it does not allow to operate with them in a type-safe way.

a) For instance:

```
plugins.withType<NexusPublishPlugin> {
  ^ Cannot access 'NexusPublishPlugin': it is public/*package*/ in 'de.marcphilipp.gradle.nexus'
```

I tend to use `allprojects {  plugins.withType<NexusPublishPlugin> {` to apply something to all the project with a given plugin

b) Then `publishToNexus` string is private: https://github.com/marcphilipp/nexus-publish-plugin/blob/3691d1fa3fdf60eb9d30461ee114c4b45f7692dc/src/main/kotlin/de/marcphilipp/gradle/nexus/NexusPublishPlugin.kt#L43

c) Then `initializeNexusStagingRepository` string is private (fortunately `tasks.withType< InitializeNexusStagingRepository>` can be used): https://github.com/marcphilipp/nexus-publish-plugin/blob/master/src/main/kotlin/de/marcphilipp/gradle/nexus/InitializeNexusStagingRepository.kt#L40

**Comment by marcphilipp** [comment]
Thanks for raising the issue. Would you have time to submit a PR?

**Comment by vlsi** [comment]
I think I can do that.

**Comment by vlsi** [comment]
`NexusPublishPlugin` is already public (it was fixed in 3ed5173b7ae0229115445f74e9f3ec51c9804b1d )
